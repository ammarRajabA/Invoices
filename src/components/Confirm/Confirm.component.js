import React, { Component } from 'react';

import Button from '../Button/Button.component'

import './Confirm.style.css'

export default class Confirm extends Component{
	render(){
		if (this.props.show===true)
			return (
				<div>
					<div className="confirm-background" onClick={this.props.onDismiss}></div>
					<div className="confirm-container">
						<span className={`confirm-header ${this.props.type}`}>
							<span className="title">{this.props.title}</span>
							<span className="close" onClick={this.props.onDismiss}>
								<i className="fas fa-times-circle" style={{color:'white'}}></i>
							</span>
						</span>
						<span className="confirm-message">{this.props.message}</span>
						<span className="confirm-buttons">
							<Button label="Cancel" type="normal" onClick={this.props.onCancel}/>
							<Button label="Confirm"type={this.props.type} onClick={this.props.onConfirm}/>
						</span>
					</div>
				</div>
				)
		else return <span></span>
	}
}

