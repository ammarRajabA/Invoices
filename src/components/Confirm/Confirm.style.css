.confirm-background{
	position: fixed;
	top:0;
	left: 0;
	width: 100%;
	height: 100%;
	background-color: #000000;
	opacity: 0.5;
	z-index: 3;
}

.confirm-container{
	display: flex;
	flex-direction: column;
	position: fixed;
	top:30%;
	right: 25%;
	background-color: #EDEDED;
	width: 50%;
	height: 40%;
	z-index: 3;
	padding-bottom: 20px;
	box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2), 0 3px 10px 0 rgba(0,0,0,0.19);
	font-family: 'calibri'
}
.confirm-container .close{
	margin-right: 10px;
	margin-left: auto;
	cursor: pointer;
	opacity: 0.8;
	transition: 0.3s
}
.confirm-container .close:hover{
	opacity: 1
}
.confirm-header{
	display: flex;
	flex-direction: row;
	align-items: center;
	width: 100%;
	height: 50px;
	font-size: 20px;
	color: white;
	background-color: #4D87EC;
}
.confirm-header .title{
	margin-left: 20px;
}
.confirm-message{
	align-self: center;
	margin-top: auto;
	margin-bottom: auto;
	font-size: 18px;
}
.confirm-buttons{
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-around;
}
.warning{
	background-color: #ffa500;
	border-color: #ee9400;
}
.warning:hover{
	background-color: #ee9400;
}

.danger{
	background-color: #D9534F;
	border-color: #C8423E;
}
.danger:hover{
	background-color: #C8423E;
}

.success{
	background-color: #61B2A7;
	border-color: #50A196;
}
.success:hover{
	background-color: #50A196;
}