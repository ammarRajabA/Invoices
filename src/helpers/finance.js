export const calculateTotalPriceWithTax=(invoiceData)=>{
	var total=0.0;
	if (Array.isArray(invoiceData)){
		invoiceData.map((invoice)=>{
			total+=calculateTotalPriceWithTax(invoice)
		})
		return total
	}else{
		invoiceData.items.map((item)=>{
			total+=item.qty*item.unitPriceNet*(1+parseFloat(item.taxRate))
		})
		return total
	}
}
export const calculateTotalPriceNoTax=(invoiceData)=>{
	var total=0.0;
	if (Array.isArray(invoiceData)){
		invoiceData.map((invoice)=>{
			total+=calculateTotalPriceWithTax(invoice)
		})
		return total
	}else{
		invoiceData.items.map((item)=>{
			total+=item.qty*item.unitPriceNet
		})
		return total
	}
}

export const calculateTotalItemNoTax=(item)=>item.qty*item.unitPriceNet
export const calculateTotalItemWithTax=(item)=>item.qty*item.unitPriceNet*(1+parseFloat(item.taxRate))