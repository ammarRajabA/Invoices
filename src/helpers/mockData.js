import faker from 'faker';

function randomInvoice() {
  const senderName = `${faker.name.firstName()} ${faker.name.lastName()}`;
  const recipientName = `${faker.name.firstName()} ${faker.name.lastName()}`;
  return {
    "id": faker.random.uuid(),
    "dueDate": faker.date.future(),
    "createdDate": faker.date.past(),
    "sender": {
      "name": senderName,
      "email": `${senderName.split(" ").map(n => n.toLowerCase()).join(".")}@sender.com`,
      "address1": `${faker.company.companyName()} ${faker.company.companySuffix()}`,
      "address2": `${faker.address.streetName()}, ${faker.random.number({min: 1, max: 100})}`,
      "address3": `${faker.address.zipCode()} ${faker.address.city()}`,
      "vatId": faker.helpers.replaceSymbolWithNumber("DE#########")
    },
    "recipient": {
      "name": recipientName,
      "email": `${recipientName.split(" ").map(n => n.toLowerCase()).join(".")}@recipient.com`,
      "address1": `${faker.company.companyName()} ${faker.company.companySuffix()}`,
      "address2": `${faker.address.streetName()}, ${faker.random.number({min: 1, max: 100})}`,
      "address3": `${faker.address.zipCode()} ${faker.address.city()}`,
      "vatId": faker.helpers.replaceSymbolWithNumber("DE#########")
    },
    "items": Array.apply(null, new Array(faker.random.number({min: 1, max: 20}))).map(() => (
      {
        "description": faker.commerce.productName(),
        "qty": faker.random.number({min: 1, max: 10}),
        "unitPriceNet": parseFloat(faker.finance.amount(0.01,100,2)),
        "taxRate": 0.19
      }
    ))
  }
}

const NUM_INVOICES = 10;
const INVOICES = Array.apply(null, new Array(NUM_INVOICES)).map(randomInvoice);

export default {
	invoices:INVOICES,
	meta:{
		total:100,
		totalPages:10,
		pageNumber:1,
		pageSize:10
	},
  INIT_STATE:{
    invoices:[],
    selectedInvoice:null,
    loading:false,
    invoiceSelected:null,
    meta:{
      total:0,
      totalPages:0,
      pageNumber:0,
      pageSize:0
    },
    newInvoice:{
      "id": -1,
        "dueDate": new Date(),
        "createdDate": new Date(),
        "sender": {
            "name": '',
            "email": '',
            "address1": '',
            "address2": '',
            "address3": '',
            "vatId": ''
        },
        "recipient": {
            "name": '',
            "email": '',
            "address1": '',
            "address2": '',
            "address3": '',
            "vatId": ''
        },
        "items": [{description:'',qty:0,unitPriceNet:0,taxRate:0}]
    }
  }
}