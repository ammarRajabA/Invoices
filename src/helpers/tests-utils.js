import React from 'react'
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

export const makeMountRender = (Component, defaultProps = {}) => {
	return (customProps={})=>{
		const props={...defaultProps,...customProps};
		return mount(<Component {...props}/>)
	}
}

export const reduxify = (Component, props = {}, store= {}) => {
	return function reduxWrap() {
		return (
			<Provider store={store}>
				<Component {...props} />
			</Provider>
		)
	}
}

export const snapshotify = reactWrapper => {
	return reactWrapper.debug();
};