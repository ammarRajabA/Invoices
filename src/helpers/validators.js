export const validateEmail=(email)=>{
	var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
}
export const validateVATID=(vatID)=>{
	return /(DE)?[0-9]{9}/.test(vatID)
}