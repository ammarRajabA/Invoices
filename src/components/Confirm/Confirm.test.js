import { makeMountRender,makeStore,reduxify, snapshotify } from '../../helpers/tests-utils';

import Confirm from './Confirm.component'

describe('Confirm component',function () {
	it('match snapshot hidden',function () {
		const wrapper = makeMountRender(Confirm)();
		expect(snapshotify(wrapper)).toMatchSnapshot();
	})
	it('match snapshot hidden',function () {
		const wrapper = makeMountRender(Confirm)({show:false});
		expect(snapshotify(wrapper)).toMatchSnapshot();
	})
	it('match snapshot hidden',function () {
		const wrapper = makeMountRender(Confirm)({show:true,onDismiss:()=>{this.setProps({show:false})}});
		wrapper.setProps({ show: false });
		expect(snapshotify(wrapper)).toMatchSnapshot();
	})
	it('match snapshot shown',function () {
		const wrapper = makeMountRender(Confirm)({show:true});
		expect(snapshotify(wrapper)).toMatchSnapshot();
	})
})