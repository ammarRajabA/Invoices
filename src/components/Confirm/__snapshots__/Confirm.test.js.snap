// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm component match snapshot hidden 1`] = `
"<Confirm>
  <span />
</Confirm>"
`;

exports[`Confirm component match snapshot hidden 2`] = `
"<Confirm show={false}>
  <span />
</Confirm>"
`;

exports[`Confirm component match snapshot hidden 3`] = `
"<Confirm show={false} onDismiss={[Function: onDismiss]}>
  <span />
</Confirm>"
`;

exports[`Confirm component match snapshot shown 1`] = `
"<Confirm show={true}>
  <div>
    <div className=\\"confirm-background\\" onClick={[undefined]} />
    <div className=\\"confirm-container\\">
      <span className=\\"confirm-header undefined\\">
        <span className=\\"title\\" />
        <span className=\\"close\\" onClick={[undefined]}>
          <i className=\\"fas fa-times-circle\\" style={{...}} />
        </span>
      </span>
      <span className=\\"confirm-message\\" />
      <span className=\\"confirm-buttons\\">
        <Button label=\\"Cancel\\" type=\\"normal\\" onClick={[undefined]}>
          <button className=\\"button normal\\" label=\\"Cancel\\" type=\\"normal\\" onClick={[undefined]}>
            <span>
              Cancel
            </span>
          </button>
        </Button>
        <Button label=\\"Confirm\\" type={[undefined]} onClick={[undefined]}>
          <button className=\\"button undefined\\" label=\\"Confirm\\" type={[undefined]} onClick={[undefined]}>
            <span>
              Confirm
            </span>
          </button>
        </Button>
      </span>
    </div>
  </div>
</Confirm>"
`;
